src/apps/backend/modules/config/config_service.py

```diff
@@ -1,77 +1,22 @@
-from modules.common.dict_util import DictUtil
-from modules.config.config_manager import ConfigManager
-from modules.config.types import DatadogConfig
+from typing import Generic, Optional, cast
 
+from modules.common.types import ErrorCode
+from modules.config.internals.config_manager import ConfigManager
+from modules.config.types import T
+from modules.error.custom_errors import MissingKeyError
 
-class ConfigService:
-    @staticmethod
-    def get_string(key: str) -> str:
-        return DictUtil.required_get_str(input_dict=ConfigManager.config, key=key)
 
-    @staticmethod
-    def get_bool(key: str) -> bool:
-        return DictUtil.required_get_bool(input_dict=ConfigManager.config, key=key)
+class ConfigService(Generic[T]):
+    config_manager: ConfigManager = ConfigManager()
 
-    @staticmethod
-    def get_db_uri() -> str:
-        return DictUtil.required_get_str(input_dict=ConfigManager.config, key="MONGODB_URI")
+    @classmethod
+    def get_value(cls, key: str, default: Optional[T] = None) -> T:
+        value: Optional[T] = cls.config_manager.get(key, default=default)
+        if value is None:
+            # Raised when key is not found in config store
```

Comment:
Kindly remove this comment.

Recommend to review the guidance here - [How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c), and do [How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c) to make sure we are adding comments accordingly.